lib/px/service/kinesis/base_request.rb

```diff
@@ -7,20 +7,20 @@
 module Px::Service::Kinesis
   class BaseRequest
     include Px::Service::Client::CircuitBreaker
-    include Px::Service::Client::Caching
 
     DEFAULT_PUT_RATE = 0.25
     FLUSH_LENGTH = 200
 
-    attr_accessor :kinesis, :stream
+    attr_accessor :stream
+    attr_reader :kinesis, :buffer
 
     # Circuit breaker configuration
     circuit_handler do |handler|
-      handler.logger =  defined?(Rails) ? Rails.logger : Logger.new(STDOUT)
+      handler.logger = nil # Or the configured logger
```

Comment:
Don't assume Rails logging.  Also, the circuit breaker is kinda spammy.  I know the other client does this this way, but it's bad too.